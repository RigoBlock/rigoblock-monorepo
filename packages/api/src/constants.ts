export enum NETWORKS {
  MAINNET = '1',
  ROPSTEN = '3',
  KOVAN = '42',
  LOCALHOST = '5777'
}

export const RPC_URLS = {
  [NETWORKS.MAINNET]: 'wss://mainnet.infura.io/ws/',
  [NETWORKS.ROPSTEN]: 'wss://ropsten.infura.io/ws/',
  [NETWORKS.KOVAN]: 'wss://kovan.infura.io/ws/',
  [NETWORKS.LOCALHOST]: 'ws://localhost:8545'
}
