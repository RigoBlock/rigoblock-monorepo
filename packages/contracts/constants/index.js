const path = require('path')

const NETWORKS = {
  ganache: 'http://localhost:8545',
  mainnet: 'https://mainnet.infura.io/v3/d48872aa1c00471c825e9d856c3c3138',
  ropsten: 'https://ropsten.infura.io/v3/d48872aa1c00471c825e9d856c3c3138',
  kovan: 'https://kovan.infura.io/v3/d48872aa1c00471c825e9d856c3c3138'
}

module.exports = {
  TMP_DIR: '.tmp',
  ARTIFACTS_DIR: path.resolve(__dirname, '../artifacts'),
  CONTRACTS_DIR: path.resolve(__dirname, '../src'),
  NETWORKS,
  GANACHE_PORT: 8545,
  GANACHE_NETWORK_ID: 5777,
  GAS_ESTIMATE: 5e6,
  CONTRACT_NAMES: [
    'AEthfinex',
    'ATotlePrimary',
    'AWeth',
    'Authority',
    'Distribution',
    'Drago',
    'DragoEventful',
    'DragoFactory',
    'DragoRegistry',
    'ERC20',
    'Exchange',
    'ExchangeEfx',
    'ExchangeV1Fork',
    'ExchangesAuthority',
    'Faucet',
    'HGetDragoData',
    'Inflation',
    'Migrations',
    'NavVerifier',
    'Owned',
    'OwnedUninitialized',
    'ProofOfPerformance',
    'RigoToken',
    'SigVerifier',
    'TokenTransferProxy',
    'TotlePrimary',
    'UnlimitedAllowanceToken',
    'Vault',
    'VaultEventful',
    'VaultFactory',
    'WETH9',
    'WrapperLock',
    'WrapperLockEth',
    'ZeroExExchangeHandler'
  ]
}
