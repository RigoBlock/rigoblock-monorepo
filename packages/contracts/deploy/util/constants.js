export const constants = {
    optimizerEnabled: false,
    timeoutMs: 30000,
    contracts: '*' as '*'
}
