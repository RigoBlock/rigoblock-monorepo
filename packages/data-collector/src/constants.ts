export const REDIS_HOST = process.env.REDIS_HOST || '127.0.0.1'
export const REDIS_PORT = parseInt(process.env.REDIS_PORT) || 6379
export const REDIS_PASSWORD = process.env.REDIS_PASSWORD || ''
export const TELEGRAF_HOST = process.env.TELEGRAF_HOST || '127.0.0.1'
export const TELEGRAF_PORT = process.env.TELEGRAF_PORT || 8125

export const TOKEN_MARKET_BASE_URL =
  'https://tokenmarket.net/blockchain/ethereum/assets'
export const CRYPTO_NEWS_BASE_URL =
  'https://cryptopanic.com/api/posts/?auth_token=5b3f861c261e7e6243a55dc7b7ca600832156886&public=true&filter=hot&currencies='
