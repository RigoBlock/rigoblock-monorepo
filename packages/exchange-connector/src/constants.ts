export const MAINNET: string = 'mainnet'
export const ROPSTEN: string = 'ropsten'
export const KOVAN: string = 'kovan'
export const RINKEBY: string = 'rinkeby'

export const PRICE_PRECISION: number = 7
export const AMOUNT_PRECISION: number = 5
export const TO_WEI: number = 1e18

export enum supportedExchanges {
  ETHFINEX = 'Ethfinex',
  ETHFINEX_RAW = 'EthfinexRaw',
  ERCDEX = 'ERCdEX',
  ZEROEXRELAYER = '0xStandardRelayer'
}

export enum NETWORKS {
  MAINNET = 1,
  KOVAN = 42,
  ROPSTEN = 3,
  RINKEBY = 4
}

export enum WS_STATUS {
  CLOSED = 'closed',
  OPEN = 'open',
  CONNECTING = 'connecting',
  CLOSING = 'closing'
}
