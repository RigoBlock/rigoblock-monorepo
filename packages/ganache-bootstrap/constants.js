module.exports = {
  GANACHE_URL: 'http://localhost:8545'
}
