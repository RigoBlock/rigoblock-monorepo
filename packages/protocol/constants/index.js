const path = require('path')

const NETWORKS = process.env.NETWORKS
  ? process.env.NETWORKS.split(',')
  : ['http://localhost:8545']

module.exports = {
  ARTIFACTS_DIR: path.resolve(__dirname, '../artifacts'),
  CONTRACTS_DIR: path.resolve(__dirname, '../solidity-contracts'),
  NETWORKS,
  GANACHE_PORT: 8545,
  GANACHE_NETWORK_ID: 5777,
  GAS_ESTIMATE: 5e6,
  CONTRACT_NAMES: [
    'Authority',
    'Distribution',
    'Drago',
    'DragoEventful',
    'DragoFactory',
    'DragoRegistry',
    'ERC20',
    'Inflation',
    'Migrations',
    'Owned',
    'OwnedUninitialized',
    'ProofOfPerformance',
    'RigoToken',
    'TokenTransferProxy',
    'UnlimitedAllowanceToken',
    'Vault',
    'VaultEventful',
    'VaultFactory',
    'WrapperLock',
    'WrapperLockEth'
  ]
}
