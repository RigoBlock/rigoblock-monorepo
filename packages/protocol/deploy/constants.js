const path = require('path')

module.exports = {
  ARTIFACTS_DIR: path.resolve(__dirname, '../artifacts'),
  CONTRACTS_DIR: path.resolve(__dirname, '../contracts'),
  NETWORKS: process.env.NETWORKS || ['http://localhost:8545'],
  GAS_ESTIMATE: 5e6
}
