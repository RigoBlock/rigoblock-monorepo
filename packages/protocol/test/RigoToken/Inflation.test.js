import { BigNumber } from 'bignumber.js'
import { GAS_ESTIMATE } from '../../constants'

const contractName = 'Inflation'
const vaultArtifact = require('../../artifacts/Vault.json')

describe('getInflationFactor', async () => {
  let vaultId
  let vaultAddress
  let value
  let group
  const inflationFactor = 1
  const fakeFactory = '0x7ce6e371085cb611fb46d5065397223ef2f000ff'
  const reward = 20
  beforeAll(async () => {
    group = baseContracts['VaultFactory'].address
    await baseContracts['VaultFactory'].createVault('test vault', 'VLT')
    const vaultData = await baseContracts['DragoRegistry'].fromSymbol('VLT')
    const [id, address] = vaultData
    vaultId = id
    vaultAddress = address
    const vaultInstance = new web3.eth.Contract(
      vaultArtifact.networks[networkId].abi,
      vaultAddress
    )
    value = web3.utils.toWei('2')
    await vaultInstance.methods.buyVault().send({
      from: accounts[1],
      value,
      gas: GAS_ESTIMATE,
      gasPrice: 1
    })
  })
  describe('setInflationFactor', () => {
    it('sets the inflation factor', async () => {
      const txHash = await baseContracts[contractName].setInflationFactor(
        group,
        inflationFactor
      )
      expect(txHash).toBeHash()
    })
    it('can only be called by the rigoblock DAO', async () => {
      await expect(
        baseContracts[contractName].setInflationFactor.sendTransactionAsync(
          group,
          inflationFactor,
          {
            // non DAO account
            from: accounts[1],
            gas: GAS_ESTIMATE,
            gasPrice: 1
          }
        )
      ).rejects.toThrowErrorMatchingSnapshot()
    })
    it("throws if we don't pass a whitelisted factory", async () => {
      await expect(
        baseContracts[contractName].setInflationFactor(
          fakeFactory,
          inflationFactor
        )
      ).rejects.toThrowErrorMatchingSnapshot()
    })
  })
  describe('getInflationFactor', () => {
    it('returns the inflation factor', async () => {
      const retrievedFactor = await baseContracts[
        contractName
      ].getInflationFactor(group)
      expect(retrievedFactor).toEqual(new BigNumber(inflationFactor))
    })
  })
  describe.skip('mintInflation', () => {
    it('allows ProofOfPerformance to mint rewards', async () => {
      const proofOfPerformance = baseContracts['ProofOfPerformance']
      const mintInflation = await baseContracts[
        contractName
      ].mintInflation.sendTransactionAsync(vaultAddress, reward, {
        from: proofOfPerformance,
        gas: GAS_ESTIMATE,
        gasPrice: 1
      })
      console.log(mintInflation)
    })
  })
  describe.skip('canWithdraw', () => {
    it('returns whether a wizard can claim reward tokens', async () => {
      const canWithdraw = await baseContracts[contractName].canWithdraw(vaultId)
      expect(canWithdraw).toBe(true)
    })
  })
  describe.skip('canWithdraw', () => {
    it('returns whether a wizard can claim reward tokens', async () => {
      const canWithdraw = await baseContracts[contractName].canWithdraw(vaultId)
      expect(canWithdraw).toBe(true)
    })
  })
})
