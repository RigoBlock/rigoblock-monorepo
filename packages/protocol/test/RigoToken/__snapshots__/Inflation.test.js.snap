// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getInflationFactor setInflationFactor can only be called by the rigoblock DAO 1`] = `"VM Exception while processing transaction: revert"`;

exports[`getInflationFactor setInflationFactor throws if we don't pass a whitelisted factory 1`] = `"VM Exception while processing transaction: revert"`;
